From f5d41598792311a1dd8d0d75f7fb9ea74c4ec5d8 Mon Sep 17 00:00:00 2001
From: Anton Blanchard <anton@samba.org>
Date: Wed, 18 Jan 2017 09:16:22 +1100
Subject: [PATCH 3/4] llvm: Work around pointer wrap issue with llist code

The llist code relies on pointer wrapping, which is undefined behaviour
in C. Details:

https://lists.ozlabs.org/pipermail/linuxppc-dev/2017-January/153054.html

We can work around it by placing the llist pointer at the start of
the struct.

Signed-off-by: Anton Blanchard <anton@samba.org>
---
 include/linux/vmalloc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/vmalloc.h b/include/linux/vmalloc.h
index 3d9d786..469b6fc 100644
--- a/include/linux/vmalloc.h
+++ b/include/linux/vmalloc.h
@@ -41,12 +41,12 @@ struct vm_struct {
 };
 
 struct vmap_area {
+	struct llist_node purge_list;    /* "lazy purge" list */
 	unsigned long va_start;
 	unsigned long va_end;
 	unsigned long flags;
 	struct rb_node rb_node;         /* address sorted rbtree */
 	struct list_head list;          /* address sorted list */
-	struct llist_node purge_list;    /* "lazy purge" list */
 	struct vm_struct *vm;
 	struct rcu_head rcu_head;
 };
-- 
2.9.3

